gen/src/generate/mod.rs

```diff
@@ -0,0 +1,38 @@
+mod char_property;
+
+mod idna;
+mod ucd;
+
+use std::fs;
+use std::io;
+use std::path::PathBuf;
+
+/// The standard auto-generated source disclaimer.
+const PREAMBLE: &'static str = "// WARNING: Auto-generated by unic-gen. DO NOT EDIT MANUALLY!";
+
+pub fn generate_all(mut crates: Vec<String>) -> io::Result<()> {
+    crates.sort();
+
+    for krate in &crates {
+        let path = tables_path(krate);
+        if path.exists() {
+            fs::remove_dir_all(&path)?;
+        }
+        fs::create_dir_all(path)?;
+    }
+
+    idna::generate_all(crates.iter().filter(|x| x.starts_with("unic-idna")))?;
```

Comment:
Sounds good. One more suggestion, then, would be to make the source/target name a positional argument, since its the only main input to the cli.

By the way, I noticed that we are assuming a one-to-one mapping between data sources (like `ucd` and `idna`) and target components for the data tables (again, `ucd` and `idna`). I think that applies to majority of cases, but there could be cases that a data source name doesn't match a target component. I think we're talking about *data source names* here, right?